### ISSUE TITLE
Support for Elgato Game Capture HD60

### ISSUE THREAD

**Issue by egee-irl** [tolga9009/elgato-gchd#15]
After following the directions in the Readme, attempting to run ./elgato-gchd results in the follow error message:

> Unable to find device. Terminating.

My device is a standard Elgato Game Capture HD60.

**Comment by tolga9009** [comment]
Hi,

unfortunately, the Elgato GameCapture HD60 is currently not supported. I'm gonna start a Kickstarter project soon for HD60 support. Stay tuned.

Cheers,
Tolga

**Comment by thistleknot** [comment]
I'd be all about elgato hd60 linux drivers!

**Comment by atom8bit** [comment]
I tried a little bit to get this to work, but to no avail. 
One thing I can offer is the HD60 (predictably) has different PIDs.  I can share mine if you like.

After using the correct PID from lsusb, I was able to get it to create the file, but OBS doesn't recognize the file as actually having a valid image.

**Comment by thistleknot** [comment]
oh no!

I was going to rely on OBS...  Well, I guess I could try VLC that supports
compression.

Actually, what I was thinking of doing, was use gpu & usb passthrough on
linux, or give up and load windows.

However, if you got the file to work.  Then maybe I will try vlc

**Comment by atom8bit** [comment]
Keep in mind my problem is that the HD60 isn't supported. Had nothing to do with OBS - VLC didn't pick it up either.

**Comment by tolga9009** [comment]
The HD60 is a completely different device and therefore currently *not* supported by this driver. Please note, that not even the new revision Elgato Game Capture HDs (non-HD60) are working with this driver, eventhough the differences are marginal.

Currently, nearly 10k lines of code are needed, to set up the device (I'm sure, I can cut it down / refactor lots of parts, but ye, it's currently 10k). If you change a *single* line of code, weird colors show up or the driver completely stops working.

The HD60 will need to be worked on from scratch, as a different hardware is used and different features are offered. The driver is extremely low-level and very specialized, therefore, there is no way for me to get the current driver working with the HD60, without having one for reverse engineering.

That beeing said, I have gained alot of experience in the process of reverse engineering the Elgato GCHD and I am sure, that I'm able to get the HD60 working under Linux aswell. Lots of people are asking for it, therefore, I've decided to start a Kickstarter campaign to give you guys this opportunity. I'm doing this for the community and I need your support.

I have lots of time on my hands over the weekend. You can expect news about the Kickstarter project tomorrow.

Cheers,
Tolga

**Comment by thistleknot** [comment]
Damn, I thought vlc could read the file?

**Comment by tolga9009** [comment]
Small update: I'm still preparing the Kickstarter project. I want the fundraising to become successful, therefore I'm preparing a video, some photos and project descriptions etc. It's taking alot of time - sorry about the delays.

But some information about the Kickstarter project:

- goal is 180€, so I can buy an Elgato Game Capture HD60 to work on it

- It will run for 30 days

- I'm expecting to finish the project somewhere between 2 weeks up to 3 months after successfully raising funds

- Should there be any problems, like a strong security mechanism or should I not be able to get it working at all, you will get full refund. There should be no risks for you.

Cheers,
Tolga

**Comment by bloodywing** [comment]
I guess you can get the funding in now time if the correct people read it. I'm thinking about the steam community which is really happy about finally getting some games for Linux. And making it possible for them to stream their let's plays would make this even more awesome.

**Comment by atom8bit** [comment]
True. If you promote this right that will be over in no time.  I'd love to stream using a Linux distro instead to be quite frank.  Be sure to post to the kickstarter, I'll chip in at least a little :)

**Comment by thistleknot** [comment]
I'll chip in too!

**Comment by egee-irl** [comment]
Add the Kickstarter link to this issue thread and you'll get it finished in no time!

**Comment by Appeltomate** [comment]
I'm totally in by spending some money on the kickstarter project when it gets started. A linux driver for the GCHD60 would be more than great and tosses me away from this crappy windows stuff on my streaming pc.

**Comment by atom8bit** [comment]
Yeah. Any news on this?

**Comment by bloodywing** [comment]
@atom8bit He needs the HD60 first and this will only happen after the kickstarting was successful. Looks like the campaign is ready yet.

**Comment by atom8bit** [comment]
@bloodywing I know. I was asking about the Kickstarter.

**Comment by tolga9009** [comment]
@atom8bit real life took over, I'm very busy at the moment. I'll keep you guys updated.

**Comment by atom8bit** [comment]
Sounds good! Looking forward to it :)

**Comment by Xaekai** [comment]
+1 to supporting your endeavors

It is critical I have linux support on the Elgato HD60 (or HD60 Pro) before the next season of **Richard and Mortimer** starts. You can't possibly understand the amount of IQ that is on the line here! I guess you'd have to already watch it to get it.

**Comment by spykes** [comment]
Any news on this? Would gladly drop a good $50 towards an HD60 Pro for you. Can also help test. Note the HD60 Pro is the internal PCIe model, however apparently it is identical to the HD60 and just uses the USB 2.0 interface of the Mozart 395S encoder rather than its native PCIe bus. Given that this is a high-end IP camera SoC there might be a good deal of documentation out there already.

**Comment by spykes** [comment]
http://www.kernellabs.com/blog/?p=3879

**Comment by tolga9009** [comment]
Hi @spykes,

we need to distinguish between the HD60 and the HD60 Pro, which are both very different devices - in terms of reverse-engineering. They may house similar / almost identical hardware, but the HD60 is a USB 2.0 device, whereas the HD60 Pro is a PCIe device.

So far, I have no experience in reverse-engineering PCIe devices. I'd need to learn from scratch. Therefore, I cannot guarantee anything. Still, I'd be happy to learn new stuff and experiment with it. The only thing I can say about the HD60 Pro is: support would not be added before HD60.

About HD60 support: I'm finally on holidays, but I'm also busy with other projects at the moment. For the Kickstarter, I'm still lacking the video. I want to take my time though and make a well-thought Kickstarter project, instead of a half-hearted one. I apologize for the delays...

**Comment by Xaekai** [comment]
I would much rather have support for the HD60 Pro. To the point where I'd be willing to furnish you one as long as you can return it when you get done / get your own. That or I'd toss $50 your way. Between spykes and I you'd already be halfway there.

**Comment by tolga9009** [comment]
@Xaekai and @spykes I will read about PCIe sniffing and evaluate, whether I can make it happen or not. Thanks for your support, it keeps me motivated doing this stuff!

**Comment by tolga9009** [comment]
@Xaekai and @spykes: during the past days, I've read about PCIe reverse-engineering and driver development and of course the HD60 Pro itself. I'm sorry for letting you guys wait and disappoint you, but I'm afraid I cannot make it happen.

This is not because I'm not believing in myself or it's impossible. It's just that it's very time consuming and I'm afraid I don't have that much freetime on my hands - just beeing realistic here. I would need to start from absolute scratch - I could not reuse anything from this project, since PCIe drivers would force me to write in kernel-space.

While developing USB in user-space using libusb / hidraw / hiddev is very mature, easy and fun, PCIe development in user-space is not; so there is no way around kernel-space. From my past projects, I've learned: writing in kernel-space is very limiting and time-consuming. One is forced to write in a subset of C (which is limiting enough and slowing progress down) and debugging is a pain (faulty function? System crash -> reboot) - it simply is not fun for me. One (or atleast I) constantly has to fight problems, which don't exist and waste lots of time.

That beeing said, there are lots of other developers out there, who enjoy writing in kernel-space and may be interested in your offer. If you look around GitHub, you might come across projects like https://github.com/patjak/bcwc_pcie, which are managed / programmed by such people. Chat with them about this, they might have interest and know-how.

I'm sorry for letting you guys hang. But it's better than empty promises. I'm gonna focus on Game Capture HD and HD60 (and maybe HD60 S) support.

Cheers,
Tolga

**Comment by mbslk280** [comment]
Likewise I'd be very happy to contribute. Newbie here. Just starting out on a HLS streaming project (just within home) and really want (need) to use Linux version of NGINX rather than Windows. Running OBS in WIndows and Ubuntu in a VM and streaming to NGINX via NAT currently, which works, but tad clunky and not the fastest, especially as i want to re-encoded multiple stream variants. Really to capture in native Ubuntu, but of course need driver for HD60 :(  And i don't think my mainframe COBOL/Assembler/etc skills are gonna help much... :)

**Comment by tolga9009** [comment]
I finally had the chance to buy a HD60. More information this weekend.

@mbslk280: currently, contributing to this project is very difficult, due to the high requirements (USB hardware analyzer). But I'm working on porting over my reverse-engineering tools over to a software-based USB capture solution.

However, since my workflow for USB 2.0 and OpenVizsla (USB hardware analyzer) is already complete, I will continue working on the HD60 with my existing tools, to quickly deliver a working driver.

**Comment by tolga9009** [comment]
From what I found out so far, no firmware extraction / upload is needed for the HD60. This means, setting up the Linux drivers will be easier. Also, it will avoid legal issues.

**Comment by tolga9009** [comment]
I will use this issue as a development diary, so everyone can keep up-to-date on this.

Main differences between HD and HD60's USB descriptors are: the HD has 5 endpoints, while the HD60 only has four. Also, the endpoints are a bit different. MPEG-TS stream is transferred over `EP 2 IN`. `EP 1 OUT` is only used during very early configuration, not sure at the moment, what's its purpose is. The other endpoints were unused during my capture session. Probably used in special cases, like firmware updating, debugging etc.

Other than that, the configuration process looks very straightforward, but not really shorter than the original HD's initialization. We have no firmware upload this time - it's now simple as sending a few thousand lines of configuration packets and then requesting bulk transfer.

Eventhough the HD and HD60 had many similarities on first sight, under the hood, they seem to be very different; atleast in terms of USB communication - I have yet to find a HD60 teardown. USB packet structures are totally different and we will need to start from scratch. Also, we don't have leaked setup scripts for the HD60 - identifying packets will be nearly impossible at this time.

**Comment by Xaekai** [comment]
Any progress @tolga9009 ?

**Comment by tolga9009** [comment]
Nope, sorry. My reverse-engineering tools didn't work for the HD60, so I had to post-pone this. It's very time-consuming and I'm currently very busy. I'll try to get something working over christmas holidays.

**Comment by fearphage** [comment]
Just checking in to see if there's any progress from the software side or the kickstarter path. I'd like to watch, assist, and/or support as best I can. Thank you for all the effort thus far.

**Comment by SindreSvendby** [comment]
Do you think you will support the  HD60S if you are able to create a driver for HD60? 
Time is always limited, but money can always help there. I will also thrown in some money to kick start this project, let us know when you are able to setup the kickstarter page :+1:  

Great work so far!

**Comment by tolga9009** [comment]
Heys and sorry about the silence,

@fearphage my plans for the HD60 didn't work out so far. Currently, I'm preparing for my final exam in April, therefore I have no time on my hands at the moment. About Kickstarter: personally, I'm not in need for Kickstarter anymore, as I've bought the HD60 on my own.

The first step is writing reverse-engineering tools to work with. A good starting point could be customizing Scott's usbparse. Then, we could compare and work our way through different captures, very similar to the approach we did with the GCHD. The HD60 is just more of a blackbox, than the GCHD already is, as we have ***absolutely*** no documentation. We don't even know, which chipsets are beeing used. But I'm sure, we'll figure out enough to get it working.

Atleast, that would be my plan. We could also think about an alternative approach and disassemble the official binaries and learn from them. Or a combination of both approaches. I'll brainstorm and gather more ideas until April and start another, fresh approach.

@SindreSvendby we're using a very generic way for capturing USB traffic under Windows. This allows us to capture with low overhead and therefore no performance loss. Personally, I've not tested USB 3.0 capturing so far, but I don't see a reason why it shouldn't work. Previously, I've ruled out HD60S support, simply due to the fact, we couldn't capture USB 3.0 traffic, without spending 5000$ on a USB 3.0 analyzer. But things have changed and I think, theoretically, there should be no problems in order to make the HD60S work.

Still, as you've pointed out, time is limited and I've already got the HD60 waiting in the queue for a *long* time. Maybe other devs might be interested in working on it in near future (looking at you, @seveirein 😜). It also depends on how difficult the project is. A USB capture might help to evaluate and get a brief overview. You can find a guide here: https://github.com/tolga9009/elgato-gchd/wiki/USB-capturing-on-Windows.

Hope this cleared up a few things.

Cheers,
Tolga

**Comment by hopperpl** [comment]
Well, out of frustration with the really horrible Elgato recording software, I started to try to implement my own USB driver/software and get rid of the Elgato's. That's how I found this project. Anyway, I did some enhanced analyzing of the GCHD and already have some new insights about a lot of configuration commands.

For example, the complete configureCommonBlockC() function can be reduced to about 5 lines as I figured out how port 0x33 mail/messages are scrambled. That applies to all mails that are sent to port 0x33.

I'm making a huge GCHD documentation with everything we know and am eager to help with the HD60 too. Enough about teasing... here's how port 0x33 messages are scrambled

XOR = (port << 5) & 0xF0  |  (port << 1) & 0x0F
foreach byte of payload starting at offset 1 { byte ^= XOR, XOR += 0x73 }

The reply is scrambled as well. When we apply this to configureCommonBlockC() and other 0x33 messages, they start to show pattern and make a bit more sense. It's made more clear in detail in the doc which I will make available within a few days. 

I'm really looking forward to it as the MB86H85 is a powerful encoder chip and Elgato only scratches the surface of its abilities. I'm certain the same applies to the HD60 as well.

**Comment by CodePanter** [comment]
@hopperpl "Elgato only scratches the surface of its abilities" sounds really promising! I would love to see this driver use the Elgato in ways the original drivers would never have been able to. I would recommend joining this project's Gitter groupchat (https://gitter.im/tolga9009/elgato-gchd) because most in depth discussions take place there.

**Comment by romanovzky** [comment]
Any news?

**Comment by capricon413** [comment]
[

  | http://gadgetlint.com/best-game-capture-card/
-- | --


](url)

**Comment by capricon413** [comment]
<a href="http://gadgetlint.com/best-game-capture-card/" target="_blank">Elgato Game capture card</a>

**Comment by 0xicl33n** [comment]
Any news at all?

**Comment by Shugabuga** [comment]
Any updates?

**Comment by bigretromike** [comment]
I would love to see that you manage to find time, or that @atom8bit found alternative or even that @hopperpl made his own drivers. But if this project is dead could you please update main page to reflect it?

**Comment by pthrasher** [comment]
Just wanted to post here to say that I've been digging into HD60s Support, and determined that it uses a capture chip from MStar Semiconductor: MST3363CNK-170

Right now I'm trying to get some documentation on that chip. They offer an entire SDK / driver for it with source from the manufacturer, but they've currently got a problem with receiving emails as I have been unable to contact them.

**Comment by egeexyz** [comment]
@pthrasher Are there other cards on the market that use that chip? Maybe some work has already been done for us

**Comment by bigretromike** [comment]
@egee-irl 

USB 3.0 Capture Device for High-Performance VGA Video - 1080p 60fps - Aluminum by StarTech 
HDMI to USB-C Video Capture Device - UVC - Plug-and-Play - Mac and Windows - 1080p by StarTech

"The capture device features USB Video Class (UVC), a video capture standard that’s natively
supported by Mac, Windows® and Linux®. Unlike other capture devices that require proprietary
driver installation, UVC based capture devices install automatically for a truly plug-and-play solution."

**Comment by bigretromike** [comment]
@pthrasher where did you find that information ? maybe HD60 also use that ?

edit: here? https://www.flickr.com/photos/ashleyvansteenacker/sets/72157676095851975/

**Comment by pthrasher** [comment]
@bigretromike yes, there. HD60 does not use that chip. HD 60 has on-board encoder because usb 2.0 isn't fast enough for the raw frames. HD60s is USB 3.0 so it has the bandwidth and doesn't need to encode frames first.

I've tried reversing the firmware... I've got some pcaps of the USB traffic... But... hehe... I have no idea what I'm doing... So I'm having to learn as I go.

**Comment by pthrasher** [comment]
Additionally... I think it is a UVC device, but it doesn't announce itself as such, so the uvc drivers don't actually kick in. It announces itself as a vendor specific device.

**Comment by bigretromike** [comment]
I have hd60 if you need any help.

**Comment by egeexyz** [comment]
> I have no idea what I'm doing... So I'm having to learn as I go.

Hell yeah, story of my life as a developer.

On a more serious note @pthrasher I also still have my hd60. Let us know if we can help you with anything :+1:

**Comment by tolga9009** [comment]
If you're new into this, I **strongly** advise you to read our group chat at https://gitter.im/tolga9009/elgato-gchd going from 27th February 2017 onward (up until end of March). We had a lot of in-depth analysis going on, including links to teardown articles (HD60, HD60 Pro).

@hopperpl also went through the trouble of creating a reverse engineering document, which is pretty much on point. I couldn't find the direct link (Google Drive link), but it should be in the group chat.

We had licensing concerns, as the laws in his country forced him to publish any reverse engineering efforts only under the requirement, of not making profit. In order to protect himself, he has published the documents under the requirement, that derivative works are not allowed to be sold.

As MIT license doesn't forbid selling (so doesn't GPL / LGPL) and I was already warned once by Elgato due to legal issues (before they were acquired by Corsair), I wasn't interested in further trouble.

That is essentially the point, where I stopped working on this project: we had a very good, in-depth reverse engineering document, however we couldn't use it, due to the license we chose for our project. It felt more like a legal court, rather than a few geeks making their beloved devices work under their beloved OS.

Nevertheless, I think you can find a lot of useful information there. Of course, you need to figure out the license issue first, if you're planning to publish your work.

Cheers,
Tolga

**Comment by bigretromike** [comment]
> This specification was written only for the private use of our audience. Any commercial use or implementation is strictly prohibited. The purpose of this specification is to allow rightful owners of the Elgato Game Capture HD device to use it without the requirement of installing the provided and limited game capture software of Elgato, and to use it on any available operating system, including many distributions of Linux as well as systems like Android or embedded architectures. The purpose of this specification further includes using the device without having to manage severe limitations and software problems as well as software malfunctions that are found in the original software on selected platforms. We are not affiliated in any way with Elgato Systems GmbH, nor do we intent to. All rights to hardware and software devices that are developed, created and distributed by Elgato are unaffected. The usage of this specification is strictly limited to private use and only to rightful owners of the hardware device thereby covered by Fair Use, stated in the United States as DMCA (17 U.S.C. § 1201 (f)) and stated in the European Union as EU Directive 2009/24 (article 6).
  
Why not use same license as the documentation then ? or @tolga9009 you wanted MIT/GPL license that much?

**Comment by tolga9009** [comment]
It isn't that easy. We have started this under MIT license, so we'd have maximum freedom and get it upstream into the Linux kernel, when it's done. The current driver design was rather a proof-of-concept / playground in order to prepare a V4L2 kernel driver. An upstream kernel driver would make the Elgato an out-of-the-box experience, rather than letting the user jump through hoops.

By publishing work under that license, your work is not compatible with GPL and therefore can't be merged to Linux. Very similar to how ZFS is license-incompatible with Linux.

**Comment by bigretromike** [comment]
so it could work as a pre-compiled module that you have to copy/compile manually. 
that way no linux distro would have it merge inside, so it wouldn't be sold with it.

Maybe it could be made as a firmware ? like the code and everything is in base code with MIT licence, but you would have to put firmware somewhere to make it work, and that firmware would be the reversed elgato thing ? something like Emulators have wheres the emulator itself is open-source etc, but the bios/firmware is part where end user need provide. 

I understand how great it would be as out-of-box solution, but as you wrote its imposible, but then most people would probably prefer 'adding firmware' by hand to make it work, than being left out.

**Comment by tolga9009** [comment]
> so it could work as a pre-compiled module that you have to copy/compile manually.
that way no linux distro would have it merge inside, so it wouldn't be sold with it.

Yepp, similar to what ZFS on Linux (https://zfsonlinux.org/) is doing. But also same limitations: tainted kernel, DKMS.

> Maybe it could be made as a firmware ? like the code and everything is in base code with MIT licence, but you would have to put firmware somewhere to make it work, and that firmware would be the reversed elgato thing ?

I get what you mean, but what's the point in that? Instead of splitting up the project into 2 parts and adding complexity, why not unify it into a single package? MIT allows sublicensing - hence I chose it for the project, to give everyone as much freedom as possible.

MIT allows you to: fork the project, sublicense the project under the new conditions and move on in a single project. Only problem: no one can re-use parts of the project published under the less permissive license in other projects; there is no way back. No chance to get this code upstream or e.g. into OBS. Still, I'm not holding anyone back - feel free to fork and move on. It's just not what I personally imagined for this project, hence I won't invest my freetime down that road.

> I understand how great it would be as out-of-box solution, but as you wrote its imposible

I said it's time consuming (for me), but not impossible. In fact, we have working code, so we aren't *that* far off actually. It's just in a horrible shape. I prefer not following hopperpl's specs down to the smallest detail, but rather sneak a peek, get some ideas, do own research / interpretation and continue working under a GPL-compliant license.

> but then most people would probably prefer 'adding firmware' by hand to make it work, than being left out.

In terms of GCHD, they'd need to "add firmware by hand" either way: it needs a proprietary blob to function. HD60 on the other hand wouldn't need any user interaction at all, as it doesn't depend on transferring the firmware / payload via USB.

**Comment by bigretromike** [comment]
> I said it's time consuming (for me), but not impossible. In fact, we have working code, so we aren't _that_ far off actually. 

for hd60 ? where, can I try it ?

**Comment by tolga9009** [comment]
I was referring to the Game Capture HD - sorry if I confused you. hopperpl's documentation is only about GCHD, not HD60. Nevertheless, it gives great insight and HD60 and GCHD are more similar than we originally thought.

The component we "talk" to, is the Fujitsu MB86H58 / MB86M03. On the HD60, it's the MB86H56A - so indeed very similar. If you look at http://www.steventoth.com/2015/2/21/elgato-60fps and compare it to the original GCHD, you will find lots of similarities (minus the analogue stuff ofc). Therefore I think it's crucial to fully understand GCHD, in order to move on to HD60.

What confused me about HD60: it had considerably more USB traffic, therefore working on it was more complex. At a later point, I figured out it most likely was the RGB light beeing programmed via USB (pulsing / wave animation). That filtered out, leaves you with a USB traffic very similar to GCHD (minus firmware loading and some other stuff). Iirc I never published this idea, as I wasn't sure about it. But if you guys want to work on this, I think that'd be interesting to check.

**Comment by bigretromike** [comment]
@tolga9009 what about Personal Use and Evaluation License (PUEL) ? ex. https://www.virtualbox.org/wiki/VirtualBox_PUEL

**Comment by diotro** [comment]
Is there any code I can slip into the *.cpp/*.h that will allow me access to the S-Video input into the DIN using the (sold separately) adapter?  I briefly scanned around the source code and didn't find any mention of it.

**BTW**, I can't thank you enough for reverse engineering these drivers. You've given me new inspiration to start brushing up my C++ skills.

**Comment by romanovzky** [comment]
You guys let me know if any advances are made that require testing :)

**Comment by Xaekai** [comment]
If you can make this work under DKMS while complying with El Gato's legal constraints, I think you should do it. Anyone who uses VirtualBox to play their Windows games already has to deal with DKMS anyway.

**Comment by raviv** [comment]
Did anyone succeed in using the HD60 as UVC device under Linux? (e.g. using opencv)

**Comment by ghost** [comment]
@tolga9009 A shame than you didnt found the support to make this true for newer elgato devices. Thank you anyway.

**Comment by S33G** [comment]
Hello folks, 
Wondering if you made any progress with this or require any input at-all?

**Comment by Xaekai** [comment]
I think he gave up.

**Comment by jonneymendoza** [comment]
Damm no update?

**Comment by ghost** [comment]
Indeed, I think we can't blame he for doing such an effort and gave up on it. Elgato should have support for linux.

**Comment by Sophira** [comment]
They already updated with the current status at https://github.com/tolga9009/elgato-gchd/issues/15#issuecomment-457807866 . Basically, @tolga9009 has stopped working on the project due to legal issues.

**Comment by egeexyz** [comment]
**tl;dr -** I'm closing this issue because the author is no longer working on this project due to licensing issues from Elgato.

If you want to work on the project yourself, please fork this repo and continue the conversation there.

I created this issue back in **2016** to start a conversation about support for the HD60. The author of the project and others began work on a solution that resulted in Elgato warning them about licensing issues. Since then, the authors decided to drop the project entirely.

There is nothing else to say or do with this issue and I'm tired of being emailed about it every few months by people who refuse to read the comment history, so I am closing it.

> Please do not comment on this issue asking for status - just scroll up and read the comments and move on.

**Comment by EpicLPer** [comment]
"Licensing" issues for an Open Source project with from ground up written drivers... what the eff is wrong with Elgato...

... then again they try to sell a glorified LED light with USB for almost 200 bucks so...

**Comment by DusteDdk** [comment]
Best thing to do is not buy Elgato stuff, they don't care about their customers.

**Comment by Askmewho** [comment]
I just sent them to hell. Thanks @tolga9009 for your effort
